src/vm.rs

```diff
@@ -586,26 +667,31 @@ impl<'a, S: Storage, C: PevmChain> Vm<'a, S, C> {
                                     || basic.balance != account.info.balance
                             })
                         {
-                            if evm.db().is_lazy {
-                                if account_location_hash == from_hash {
-                                    write_set.push((
-                                        account_location_hash,
-                                        MemoryValue::LazySender(U256::MAX - account.info.balance),
-                                    ));
-                                } else if Some(account_location_hash) == to_hash {
+                            match evm.db().lazy_strategy {
+                                LazyStrategy::RawTransfer => {
+                                    if account_location_hash == from_hash {
+                                        write_set.push((
+                                            account_location_hash,
+                                            MemoryValue::LazySender(
+                                                U256::MAX - account.info.balance,
+                                            ),
+                                        ));
+                                    } else if Some(account_location_hash) == to_hash {
+                                        write_set.push((
+                                            account_location_hash,
+                                            MemoryValue::LazyRecipient(tx.value),
+                                        ));
+                                    }
+                                }
+                                LazyStrategy::ERC20Transfer { .. } | LazyStrategy::None => {
```

Comment:
A lazy ERC-20 transfer with zero ETH value can skip a write to the recipient basic here? Perhaps we can also apply the same zero ETH value check to raw transfer. 

We can also `LazySender` the sender here? I.e., we can mock the sender basic if it's interacting with ERC-20, too.

Reply:
good idea!